core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala

```diff
@@ -253,6 +265,32 @@ class ResourceProfile(
   }
 }
 
+/**
+ * Resource profile which only contains task resources, used for stage level task schedule when
+ * dynamic allocation is disabled, tasks will be scheduled to executors with default resource
+ * profile based on task resources described by this task resource profile.
+ *
+ * @param taskResources Resource requests for tasks. Mapped from the resource
+ *                      name (e.g., cores, memory, CPU) to its specific request.
+ */
+@Evolving
+@Since("3.4.0")
+class TaskResourceProfile(override val taskResources: Map[String, TaskResourceRequest])
+  extends ResourceProfile(Map.empty, taskResources) {
```

Comment:
How about:
```suggestion
  extends ResourceProfile(ResourceProfile.getOrCreateDefaultProfile(SparkEnv.get.conf).executorResources, taskResources) {
```
so we don't need to override `getTargetCustomExecutorResources()`?

Reply:
Thanks, make sense. Also we can leverage the idea `reuse compatible executors` in this way.